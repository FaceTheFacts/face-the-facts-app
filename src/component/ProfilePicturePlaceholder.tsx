import React from 'react';
import Svg, {Circle, Path} from 'react-native-svg';

const ProfilePicturePlaceholder = () => {
  return (
    <Svg width={76} height={76} viewBox="0 0 76 76">
      <Circle cx={38} cy={38} r={30} fill="white" />
      <Path
        d="M38,0c-21.078,0 -38,17.07 -38,38c0,21.078 16.922,38 38,38c20.93,0 38,-16.922 38,-38c0,-20.93 -17.07,-38 -38,-38Zm0,19c5.789,0 10.687,4.898 10.687,10.687c0,5.938 -4.898,10.688 -10.687,10.688c-5.937,0 -10.687,-4.75 -10.687,-10.688c-0,-5.789 4.75,-10.687 10.687,-10.687Zm0,47.5c-7.867,0 -14.992,-3.117 -20.187,-8.313c2.374,-6.234 8.312,-10.687 15.437,-10.687l9.5,0c6.977,0 12.914,4.453 15.289,10.687c-5.195,5.196 -12.32,8.313 -20.039,8.313Z"
        fill="#181924"
      />
    </Svg>
  );
};

export default ProfilePicturePlaceholder;
