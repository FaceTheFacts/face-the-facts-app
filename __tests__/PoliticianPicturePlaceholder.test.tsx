jest.mock('react-native-fs', () => ({
  CachesDirectoryPath: jest.fn(),
  downloadFile: jest.fn(),
  exists: jest.fn(),
  ExternalStorageDirectoryPath: 'package-path',
}));

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProfilePicturePlaceholder from '../src/component/ProfilePicturePlaceholder';

it('PoliticianPicture renders correctly', () => {
  const tree = renderer.create(<ProfilePicturePlaceholder />).toJSON();
  expect(tree).toMatchSnapshot();
});
