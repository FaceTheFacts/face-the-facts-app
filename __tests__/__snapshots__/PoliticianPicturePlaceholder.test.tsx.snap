// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoliticianPicture renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={76}
  bbWidth={76}
  focusable={false}
  height={76}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 76,
        "width": 76,
      },
    ]
  }
  vbHeight={76}
  vbWidth={76}
  width={76}
>
  <RNSVGGroup>
    <RNSVGCircle
      cx={38}
      cy={38}
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
      r={30}
    />
    <RNSVGPath
      d="M38,0c-21.078,0 -38,17.07 -38,38c0,21.078 16.922,38 38,38c20.93,0 38,-16.922 38,-38c0,-20.93 -17.07,-38 -38,-38Zm0,19c5.789,0 10.687,4.898 10.687,10.687c0,5.938 -4.898,10.688 -10.687,10.688c-5.937,0 -10.687,-4.75 -10.687,-10.688c-0,-5.789 4.75,-10.687 10.687,-10.687Zm0,47.5c-7.867,0 -14.992,-3.117 -20.187,-8.313c2.374,-6.234 8.312,-10.687 15.437,-10.687l9.5,0c6.977,0 12.914,4.453 15.289,10.687c-5.195,5.196 -12.32,8.313 -20.039,8.313Z"
      fill={4279769380}
      propList={
        Array [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
